src/validation.cpp

```diff
@@ -3431,7 +3431,7 @@ bool CChainState::AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CVali
     return true;
 }
 
-bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<const CBlock> pblock, bool fForceProcessing, bool *fNewBlock)
+bool ProcessNewBlock(const CChainParams& chainparams, const std::shared_ptr<const CBlock> pblock, bool fForceProcessing, bool *fNewBlock, const CDiskBlockPos *diskpos, const bool do_ooob)
```

Comment:
Instead of refactoring PNB to pass the diskpos through, I think it would be more maintainable if you just call AcceptBlock from outoforder.cpp.

Reply:
AcceptBlock is not exposed outside of validation.cpp, and calling it directly would entail rewriting the rest of ProcessNewBlock too - which will make rebasing more complex, as any changes there would need to be ported over.